### PR TITLE
Add PageableOperation to Core

### PR DESCRIPTION
**Pull request by maririos**
Both Text Analytics and Document Translator have been using the `PageableOperation` class for more than 2 releases. We did user studies in both languages and got positive feedback on the implementation of the `PageableOperation`. 
The users did request an improvement in the documentation so my idea is for each library to add a link to the `Azure.Core` sample for Pageable Operation.

Related: https://github.com/Azure/azure-sdk-for-net/issues/20412

### CODE REVIEW

File: sdk/core/Azure.Core/samples/LongRunningOperations.md [hunk 1]
```diff
@@ -38,3 +38,72 @@ Console.WriteLine(operation.HasCompleted);
...
+A Pageable Operation is use when the service call returns multiple values in pages after the Long Running Operation completes. The results can be access with the `GetValues()`, `GetValuesAsync()` methods which return `Pageable<T>/AsyncPageable<T>` respectively.
+
+To access the result you can iterate over the `Pageable<T>/AsyncPageable<T>`. for more information see [Consuming Service Methods Returning Pageable/AsyncPageable](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/core/Azure.Core#consuming-service-methods-returning-asyncpageablet).
+
+### Using `Value` property
+The `Value` property will contain the `AsyncPageable<T>` results.
+
+```C#
+// create a client
+var client = new TextAnalyticsClient(new Uri("http://example.com"), new DefaultAzureCredential());
```
Comment (maririos) [inline sdk/core/Azure.Core/samples/LongRunningOperations.md:53]: Ideally, these will be code snippets. I need to change TA to target Core with a project reference so I can actually do it

File: sdk/core/Azure.Core/samples/LongRunningOperations.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 **NOTE:** Samples in this file apply only to packages that follow [Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html). Names of such packages usually start with `Azure`. 
 
-Some operations take long time to complete and require polling for their status. Methods starting long-running operations return `*Operation<T>` types.
+Some operations take long time to complete and require polling for their status. Methods starting long-running operations return either `*Operation<T>` or `PageableOperation<T>` types.
```
Comment (pakrym) [inline sdk/core/Azure.Core/samples/LongRunningOperations.md:5]: Would they return a `PageableOperation<T>` directly or a subtype of it?

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs [hunk 3]
```diff
@@ -331,7 +331,7 @@ async ValueTask<OperationState<AsyncPageable<AnalyzeHealthcareEntitiesResultColl
                 AnalyzeHealthcareEntitiesResultCollection value = Transforms.ConvertToAnalyzeHealthcareEntitiesResultCollection(response.Value.Results, _idToIndexMap);
                 _firstPage = Page.FromValues(new List<AnalyzeHealthcareEntitiesResultCollection>() { value }, nextLink, rawResponse);
 
-                return OperationState<AsyncPageable<AnalyzeHealthcareEntitiesResultCollection>>.Success(rawResponse, CreateOperationValueAsync());
+                return OperationState<AsyncPageable<AnalyzeHealthcareEntitiesResultCollection>>.Success(rawResponse, CreateOperationValueAsync(cancellationToken));
```
Comment (pakrym) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs:334]: I don't think it's correct to use this `cancellationToken` here. This one would be for the `UpdateStateAsync` operation we don't want to capture it inside an IAsyncEnumerable.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs [hunk 4]
```diff
@@ -258,15 +258,15 @@ public virtual async Task CancelAsync(CancellationToken cancellationToken = defa
         /// <remarks>
         /// Operation must complete successfully (HasValue is true) for it to provide values.
         /// </remarks>
-        public override AsyncPageable<AnalyzeHealthcareEntitiesResultCollection> GetValuesAsync() => _operationInternal.Value;
+        public override AsyncPageable<AnalyzeHealthcareEntitiesResultCollection> GetValuesAsync(CancellationToken cancellationToken = default) => _operationInternal.Value;
```
Comment (pakrym) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs:261]: We might need to call `CreateOperationValueAsync` here and use the provided token.

File: sdk/core/Azure.Core/samples/LongRunningOperations.md (hunk 1 above)
Comment (pakrym) [reply sdk/core/Azure.Core/samples/LongRunningOperations.md:53]: Let's add a project reference for now and flip to a package reference when shipped.

File: sdk/core/Azure.Core/samples/LongRunningOperations.md (hunk 2 above)
Comment (maririos) [reply sdk/core/Azure.Core/samples/LongRunningOperations.md:5]: Good point.  I will update to `*PageableOperation<T>`

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs (hunk 4 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs:261]: ahh thank you. I was confused here

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs (hunk 3 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs]: removed

File: sdk/core/Azure.Core/samples/LongRunningOperations.md (hunk 1 above)
Comment (maririos) [reply sdk/core/Azure.Core/samples/LongRunningOperations.md:53]: Hmm how can I avoid the cycle dependency between TA and Core?
Currently, I have TA targeting Core so we can test that the changes added to Core are working. In theory I can undo this and only take the change once Core ships, but prefer not to

Comment (pakrym) [reply sdk/core/Azure.Core/samples/LongRunningOperations.md:53]: I don't think there is a cycle

```
TA -> Core
Core.Tests -> TA -> Core
Core.Tests -> Core
```

Comment (maririos) [reply sdk/core/Azure.Core/samples/LongRunningOperations.md:53]: Changos! Azure.Core.Tests... you are right. it should be ok.
Will update

**Review by pakrym — APPROVED** [review]

**Pull request merged**